src/Type/Php/DateFunctionReturnTypeHelper.php

```diff
@@ -76,7 +76,7 @@ public function buildReturnTypeFromFormat(string $formatString, bool $useMicrose
 				return $this->buildNumericRangeType(0, 1, false);
 			case 'u':
 				return $useMicrosec
-					? new IntersectionType([new StringType(), new AccessoryNonFalsyStringType()])
+					? new IntersectionType([new StringType(), new AccessoryNumericStringType()])
```

Comment:
Please keep the non-falsy type as well. Also, you could add type inference for `v`.

Reply:
done